zebra-network/src/peer_set/set/tests/prop.rs

```diff
@@ -90,17 +96,85 @@ proptest! {
             Ok::<_, TestCaseError>(())
         })?;
     }
+
+    /// Test if requests are broadcasted to the right number of peers.
+    #[test]
+    fn broadcast_to_peers(
+        peer_versions in any::<PeerVersions>(),
+    ) {
+        // Get a dummy block hash to help us construct a valid request to be broadcasted
+        let block: block::Block = zebra_test::vectors::BLOCK_MAINNET_10_BYTES
+            .zcash_deserialize_into()
+            .unwrap();
+        let block_hash = block::Hash::from(&block);
+
+        // Get the current valid peer version
+        let current_mainnet_version = Version::min_specified_for_upgrade(
+            Network::Mainnet,
+            NetworkUpgrade::Canopy
+        );
+
+        // Start the runtime
+        let runtime = zebra_test::init_async();
+        let _guard = runtime.enter();
+
+        // Get peers and handles
+        let (discovered_peers, mut handles) = peer_versions.mock_peer_discovery();
+        let (minimum_peer_version, _best_tip_height) =
+            MinimumPeerVersion::with_mock_chain_tip(Network::Mainnet);
+
+        // Build a peerset
+        runtime.block_on(async move {
+            let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
+                .with_discover(discovered_peers)
+                .with_minimum_peer_version(minimum_peer_version.clone())
+                .build();
+
+            // Get the total number of active peers
+            let total_number_of_active_peers = check_if_only_up_to_date_peers_are_live(
+                &mut peer_set,
+                &mut handles,
+                current_mainnet_version,
+            )?;
+
+            // Exit the test early if the total number of active peers is zero
+            if total_number_of_active_peers == 0 {
+                return Ok::<_, TestCaseError>(());
+            }
```

Comment:
How can we make sure that the rest of the test gets run for some test cases? What if the test has a bug, and we always exit at this line?

Since the peer version isn't important for this test, we could just pick a random number of peers between `1..100`, and give them all `CURRENT_NETWORK_PROTOCOL_VERSION`. Then we can assert that we get exactly that number of peers each time.